x/auth/keeper/deterministic_test.go

```diff
@@ -118,3 +153,363 @@ func pubkeyGenerator(t *rapid.T) *rapid.Generator[secp256k1.PubKey] {
+	queryClient := suite.createAndReturnQueryClient(ak)
+	for i := 0; i < iterCount; i++ {
+		res, err := queryClient.ModuleAccounts(suite.ctx, &types.QueryModuleAccountsRequest{})
+		suite.Require().NoError(err)
+		suite.Require().NotNil(res)
+		suite.Require().NotNil(res.Accounts)
+		suite.Require().Len(res.Accounts, len(prevRes))
+
+		unpackedAccs := make([]types.AccountI, len(res.Accounts))
+		for i := 0; i < len(res.Accounts); i++ {
+			var account types.AccountI
+			err = suite.encCfg.InterfaceRegistry.UnpackAny(res.Accounts[i], &account)
+			suite.Require().NoError(err)
+
+			unpackedAccs[i] = account
+		}
+
+		if prevRes != nil {
+			for i := 0; i < len(prevRes); i++ {
+				suite.Require().Equal(unpackedAccs[i].GetAddress(), prevRes[i].GetAddress())
+				suite.Require().Equal(unpackedAccs[i].GetAccountNumber(), prevRes[i].GetAccountNumber())
+				suite.Require().Equal(unpackedAccs[i].GetSequence(), prevRes[i].GetSequence())
+			}
+		}
+
+		prevRes = unpackedAccs
+	}
+
+	for i := 0; i < len(prevRes); i++ {
+		suite.accountKeeper.RemoveAccount(suite.ctx, prevRes[i])
+	}
+}
+
+func (suite *DeterministicTestSuite) TestGRPCQueryModuleAccounts() {
+	permissions := []string{"burner", "minter", "staking", "random"}
+
+	rapid.Check(suite.T(), func(t *rapid.T) {
+		maccsCount := rapid.IntRange(1, 10).Draw(t, "accounts")
+		maccs := make([]string, maccsCount)
+
+		for i := 0; i < maccsCount; i++ {
+			maccs[i] = rapid.StringMatching(`[a-z]{5,}`).Draw(t, "module-name")
+		}
+
+		maccPerms := make(map[string][]string)
+		for i := 0; i < maccsCount; i++ {
+			mPerms := make([]string, 0, 4)
+			for _, permission := range permissions {
+				if rapid.Bool().Draw(t, "permissions") {
+					mPerms = append(mPerms, permission)
+				}
+			}
+
+			if len(mPerms) == 0 {
+				num := rapid.IntRange(0, 3).Draw(t, "num")
+				mPerms = append(mPerms, permissions[num])
+			}
+
+			maccPerms[maccs[i]] = mPerms
+		}
+
+		ak := keeper.NewAccountKeeper(
+			suite.encCfg.Codec,
+			suite.key,
+			types.ProtoBaseAccount,
+			maccPerms,
+			"cosmos",
+			types.NewModuleAddress("gov").String(),
+		)
+
+		storedMaccs := suite.setModuleAccounts(suite.ctx, ak, maccs)
+
+		suite.runModuleAccountsIterations(ak, storedMaccs)
+	})
+
+	maccPerms := map[string][]string{
+		"fee_collector":          nil,
+		"mint":                   {"minter"},
+		"bonded_tokens_pool":     {"burner", "staking"},
+		"not_bonded_tokens_pool": {"burner", "staking"},
+		multiPerm:                {"burner", "minter", "staking"},
+		randomPerm:               {"random"},
+	}
+
+	ak := keeper.NewAccountKeeper(
+		suite.encCfg.Codec,
+		suite.key,
+		types.ProtoBaseAccount,
+		maccPerms,
+		"cosmos",
+		types.NewModuleAddress("gov").String(),
+	)
+
+	maccs := make([]string, 0, len(maccPerms))
+	for k := range maccPerms {
+		maccs = append(maccs, k)
+	}
+
+	sort.Strings(maccs)
+	storedMaccs := suite.setModuleAccounts(suite.ctx, ak, maccs)
```

Comment:
same as for the bech32 test: for regression we can simply use suite.ak, what do you think?

Reply:
yeah, that's a better idea!